### PR TITLE
chore(en/superstream): enable NSFW & (en/nollyverse): added missing refactor

### PR DESCRIPTION
**Pull request by fakebhai**
Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [X] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [X] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/en/nollyverse/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ext {
     extName = 'NollyVerse'
     extClass = '.NollyVerse'
-    extVersionCode = 3
+    extVersionCode = 4
```
Comment (Secozzi) [inline src/en/nollyverse/build.gradle:4]: Why are you bumping when it's just a small code refactor?

File: src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStreamAPI.kt [hunk 2]
```diff
@@ -718,7 +718,7 @@ class SuperStreamAPI(val json: Json) {
     private val client = configureToIgnoreCertificate()
 
     // 0 to get nsfw
-    private val hideNsfw = 1
+    private val hideNsfw = 0
```
Comment (Secozzi) [inline src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStreamAPI.kt:721]: I'd say make this toogleable in extension settings.

File: src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStream.kt [hunk 3]
```diff
@@ -31,14 +33,14 @@ class SuperStream : ConfigurableAnimeSource, AnimeHttpSource() {
 
     private val json: Json by injectLazy()
 
-    private val superStreamAPI = SuperStreamAPI(json)
-
-    override val baseUrl = superStreamAPI.apiUrl
-
     private val preferences: SharedPreferences by lazy {
         Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
     }
```
Comment (Secozzi) [inline src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStream.kt:38]: There's no need to use lazy if you're going to the preferences directly after

File: src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStreamAPI.kt [hunk 4]
```diff
@@ -713,12 +714,12 @@ const val TYPE_SERIES = 2
...
 
 // Ported from CS3
-class SuperStreamAPI(val json: Json) {
+class SuperStreamAPI(val json: Json, preferences: SharedPreferences) {
 
     private val client = configureToIgnoreCertificate()
 
     // 0 to get nsfw
-    private val hideNsfw = 1
+    private var hideNsfw = if (preferences.getBoolean(SuperStream.PREF_HIDE_NSFW_KEY, SuperStream.PREF_HIDE_NSFW_DEFAULT)) 1 else 0
```
Comment (Secozzi) [inline src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStreamAPI.kt:722]:
```suggestion
    private val hideNsfw = if (preferences.getBoolean(SuperStream.PREF_HIDE_NSFW_KEY, SuperStream.PREF_HIDE_NSFW_DEFAULT)) 1 else 0
```

File: src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStream.kt [hunk 5]
```diff
@@ -187,8 +187,15 @@ class SuperStream : ConfigurableAnimeSource, AnimeHttpSource() {
...
     override fun animeDetailsParse(response: Response) = throw UnsupportedOperationException()
 
+    companion object {
+        const val PREF_HIDE_NSFW_KEY = "pref_hide_nsfw"
+        const val PREF_HIDE_NSFW_DEFAULT = true
```
Comment (Secozzi) [inline src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStream.kt:192]: Make these private

File: src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStreamAPI.kt [hunk 6]
```diff
@@ -713,12 +714,12 @@ const val TYPE_SERIES = 2
...
 
 // Ported from CS3
-class SuperStreamAPI(val json: Json) {
+class SuperStreamAPI(val json: Json, preferences: SharedPreferences) {
 
     private val client = configureToIgnoreCertificate()
 
     // 0 to get nsfw
-    private val hideNsfw = 1
+    private val hideNsfw = if (preferences.getBoolean(SuperStream.PREF_HIDE_NSFW_KEY, SuperStream.PREF_HIDE_NSFW_DEFAULT)) 1 else 0
```
Comment (Secozzi) [inline src/en/superstream/src/eu/kanade/tachiyomi/animeextension/en/superstream/SuperStreamAPI.kt:722]: Create a new val for `preferences.hideNsfw` in the SuperStream main class

**Review by jmir1 — APPROVED** [review]

**Pull request merged**